backend/pkg/elasticsearch/client/augur_client_queries.go

```diff
@@ -0,0 +1,323 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/Avi18971911/Augur/pkg/elasticsearch/model"
+	"io"
+	"log"
+	"strings"
+	"time"
+)
+
+const timeOut = 1 * time.Second
+
+func (a *AugurClientImpl) Search(
+	ctx context.Context,
+	query string,
+	indices []string,
+	queryResultSize *int,
+) ([]map[string]interface{}, error) {
+	res, err := a.es.Search(
+		a.es.Search.WithContext(ctx),
+		a.es.Search.WithIndex(indices...),
+		a.es.Search.WithBody(strings.NewReader(query)),
+		a.es.Search.WithPretty(),
+		a.es.Search.WithSize(getQuerySize(queryResultSize)),
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute query: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return nil, fmt.Errorf("failed to execute query: %s", res.String())
+	}
+
+	var esResponse model.EsResponse
+	if err := json.NewDecoder(res.Body).Decode(&esResponse); err != nil {
+		return nil, fmt.Errorf("failed to decode response body: %w", err)
+	}
+
+	var results []map[string]interface{}
+	for _, hit := range esResponse.Hits.HitArray {
+		results = append(results, hit.Source)
+		results[len(results)-1]["_id"] = hit.ID
+	}
+
+	return results, nil
+}
+
+func (a *AugurClientImpl) Count(
+	ctx context.Context,
+	query string,
+	indices []string,
+) (int64, error) {
+	res, err := a.es.Count(
+		a.es.Count.WithContext(ctx),
+		a.es.Count.WithIndex(indices...),
+		a.es.Count.WithBody(strings.NewReader(query)),
+		a.es.Count.WithPretty(),
+	)
+
+	if err != nil {
+		return 0, fmt.Errorf("failed to execute query: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return 0, fmt.Errorf("failed to execute query: %s", res.String())
+	}
+
+	var countResponse model.CountResponse
+	if err := json.NewDecoder(res.Body).Decode(&countResponse); err != nil {
+		return 0, fmt.Errorf("failed to decode response body: %w", err)
+	}
+
+	return int64(countResponse.Count), nil
+}
+
+func (a *AugurClientImpl) UpdateByQuery(
+	ctx context.Context,
+	query string,
+	indices []string,
+) error {
+	res, err := a.es.UpdateByQuery(
+		indices,
+		a.es.UpdateByQuery.WithBody(strings.NewReader(query)),
+		a.es.UpdateByQuery.WithContext(ctx),
+		a.es.UpdateByQuery.WithRefresh(a.refreshRate == string(Immediate)),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to update by query in Elasticsearch: %w", err)
+	}
+	defer res.Body.Close()
+	if res.IsError() {
+		return fmt.Errorf("update by query error: %s", res.String())
+	}
+	return nil
+}
+
+type SearchAfterParams struct {
+	CreatedAt time.Time
+}
+
+type SearchAfterResult struct {
+	Success *SearchAfterSuccess
+	Error   *error
+}
+
+type SearchAfterSuccess struct {
+	CreatedAt time.Time
+	Result    []map[string]interface{}
+}
+
+func (a *AugurClientImpl) SearchAfter(
```

Comment:
This is actually new, and very long. Allows pagination of a search using a PIT and Go channel